TYP: Remove ``None`` from definition of ``DTypeLike`` type alias (#29739)
numpy/numpy: doc/release/upcoming_changes/29739.change.rst

> ``DTypeLike`` ``TypeAlias`` no longer includes ``None``

Reviewer: ```suggestion
``numpy.typing.DTypeLike`` no longer accepts ``None``
```

-------------------------------------------------------

> The type alias ``DTypeLike`` no longer includes ``None``. Instead of

Reviewer: ```suggestion
The type alias ``numpy.typing.DTypeLike`` no longer accepts ``None``. Instead of
```

.. code-block:: python

>     var: DTypeLike = None

Reviewer: usually params like these are named `dtype`:

```suggestion
    dtype: DTypeLike = None
```

> you will need to type

Reviewer: ```suggestion
it should now be
```

.. code-block:: python

>     var: DTypeLike | None = None

Reviewer: ```suggestion
    dtype: DTypeLike | None = None
```
